src/components/Card.tsx

```diff
@@ -0,0 +1,128 @@
+import {
+  SwipeableListItem,
+  Type,
+  SwipeAction,
+  TrailingActions,
+} from "react-swipeable-list";
+import { format, isToday } from "date-fns";
+import { de } from "date-fns/locale";
+import clsx from "clsx";
+import { Transition } from "@headlessui/react";
+import { useState } from "react";
+import "react-swipeable-list/dist/styles.css";
+import { Trash, Bookmark } from "react-feather";
+
+type Props = {
+  data: {
+    title?: string;
+    dateCreated: Date;
+    checkedItems: number;
+    totalItems: number;
+  };
+  createNewCard?: Boolean;
+  favorite?: Boolean;
+  onRemove: Function;
+};
+
+export default function Card({
+  data: { dateCreated, checkedItems, totalItems, title },
+  createNewCard,
+  favorite,
+  onRemove,
+}: Props) {
+  const [swiped, setSwiped] = useState(true);
+  const [inDeletion, setIndDeletion] = useState(false);
+  const [cardBackgroundSwipe, setCardBackgroundSwipe] = useState("");
+
+  function deleteListItem() {
+    setIndDeletion(true);
+    setSwiped(false);
+    onRemove();
+  }
+
+  const trailingActions = () => (
+    <TrailingActions>
+      <SwipeAction onClick={() => deleteListItem()}>
+        <DeleteList />
+      </SwipeAction>
+    </TrailingActions>
+  );
+
+  function styleSwipe(direction: string) {
+    setCardBackgroundSwipe(direction);
+  }
+
+  const basicCardStyle = "rounded-2xl  w-full h-44 ";
+
+  return (
+    <Transition
+      show={swiped}
+      appear={true}
+      enter="transform transition duration-[400ms]"
+      enterFrom="opacity-0 scale-y-0"
+      enterTo="opacity-100 scale-y-100"
+      leave="transform duration-[400ms] transition ease-in-out"
+      leaveFrom="opacity-100 scale-y-100"
+      leaveTo="opacity-0 scale-y-0"
+    >
+      <div
+        className={clsx(
+          basicCardStyle,
+          "-z-10 absolute",
+          cardBackgroundSwipe === "left" && "bg-ux-error",
+          cardBackgroundSwipe === "right" && "bg-ux-success"
+        )}
+        id="swipe-bg"
+      ></div>
+      <SwipeableListItem
+        listType={Type.IOS}
+        trailingActions={trailingActions()}
```

Comment:
does trailingActions need to be a callable function? 
Couldn't this also just reference a binding?

Reply:
refactored in db9c0c6